src/game.cpp

```diff
@@ -7968,10 +7984,122 @@ bool pet_menu(monster *z)
 
         return true;
     }
+    if ( inject == choice ) {
+        auto filter = [] (const item &it) {
+            for (auto cont : it.contents) {
+                return cont.has_flag( "MUTAGEN_CEPHALOPOD" ) || cont.has_flag( "PURIFIER" );
```

Comment:
It will work right now because everything has at most 1 contained item, but it wouldn't if something could have more than one.